frontend/src/Components/PickupShoppingList/GroceryList.js

```diff
@@ -0,0 +1,142 @@
+import React from 'react';
+import './PickupShoppingList.css';
+import { findDOMNode } from "react-dom";
+
+import {Container, Row, Col, Button, InputGroup,FormControl} from 'react-bootstrap';
+
+class GroceryList extends React.Component {
+
+    constructor(props){
+        super(props)
+        this.state = {
+            newItem: "",
+            currentItems: []
+        }
+
+        this.handleChange = this.handleChange.bind(this);
+        this.handleAddItem = this.handleAddItem.bind(this);
+        this.handleDeleteItem = this.handleDeleteItem.bind(this);
+
+        this.scrollableShoppingList = null;
+    }
+
+    componentDidUpdate(){
+        const itemToFocusOn = findDOMNode(this.scrollableShoppingList);
+        // to force the table to scroll to the bottom of the table where the newest element has been added (each item row is approx 45 pixels tall)
+        itemToFocusOn.scrollTop = (this.state.currentItems.length)*45;
+    }
+
+    handleChange(event){
+        this.setState({ newItem: (event && event.target && event.target.value) || "" });
+    }
+
+    validateNewItem(newItem){
+
+        const error =  {}
+        error.foundErr = false;
+        error.errMessage = "";
+
+        // Check for repeats (can make more complex in the future)
```

Comment:
Is there any quantity feature? If not, does that make tracking repeats intrusive?

Reply:
I think for the quantity the user can list it in their entry (i.e. 5 bananas). This check was primarily to help the customer not reiterate themselves, like if they forget the items they already entered (i.e. customer enters `milk` and they try to enter `Milk` again).